### DISCUSSION TITLE
How to use tabster with Fluent UI v9?

### DISCUSSION THREAD

**Discussion by AkiraVoid** [microsoft/fluentui#27883] [Q&A] [answered]
Fluent UI v9's doc said it uses [tabster](https://github.com/microsoft/tabster) to implement keyboard navigation, but it haven't said how it uses tabster. We have some custom components which we want them to be keyboard-navigable, but we don't know how. Tabster's doc said how to use its APIs, but we think Fluent UI has created a wrapper for it. We found some hooks like: `useArrowNavigationGroup`, `useFocusableGroup`, but none of them worked.

**Comment by layershifter** [comment, marked as answer]
There is a set of hooks exported by `@fluentui/react-components`/`@fluentui/react-tabster`:

https://github.com/microsoft/fluentui/blob/ab53a4c86a49a4d4268f3d87809bee263f818c5d/packages/react-components/react-tabster/src/index.ts#L2-L9

Here is a simple example for `useArrowNavigationGroup`:

```tsx
export default function App() {
  const attrs = useArrowNavigationGroup({ axis: "horizontal", circular: true });

  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
      <h2>Start editing to see some magic happen!</h2>

      <div style={{ display: "flex", gap: 10 }}>
        <button>before</button>
        <div {...attrs} style={{ border: "4px dotted grey", padding: 4 }}>
          <button>button 1</button>
          <button>button 2</button>
          <button>button 3</button>
          <button>button 4</button>
        </div>
        <button>after</button>
      </div>
    </div>
  );
}
```

https://codesandbox.io/s/wonderful-farrell-r94kbe?file=/src/App.js:92-709

---

However, I agree. As we recommend to use our wrapper, we should have docs for it. @AkiraVoid would you mind to create an issue for that?

**Reply by AkiraVoid** [reply]
I created #27961 for this discussion.
